src/main/java/org/datadog/jmxfetch/Instance.java

```diff
@@ -83,10 +84,16 @@ public Instance(LinkedHashMap<String, Object> instanceMap, LinkedHashMap<String,
             // a jmxtree that is not completely initialized and would be missing some attributes
         }
 
-        this.minCollectionPeriod = (Integer) instanceMap.get("min_collection_interval");
+        this.minCollectionPeriod = (Integer) this.instanceMap.get("min_collection_interval");
         if (this.minCollectionPeriod == null && initConfig != null) {
         	this.minCollectionPeriod = (Integer) initConfig.get("min_collection_interval");
         }
+
+        this.emptyDefaultHostname = (Boolean) this.instanceMap.get("empty_default_hostname");
```

Comment:
Again, not your fault as you're following a pattern used in the class, but I'd say it's safer to use `boolean` as the type for `this.emptyDefaultHostname`.